cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/Joiner.java

```diff
@@ -17,26 +17,48 @@
 package io.cdap.cdap.etl.api;
 
 import io.cdap.cdap.api.annotation.Beta;
+import org.apache.commons.lang.NotImplementedException;
+
+import java.util.Collection;
+import java.util.Collections;
 
 /**
  * Provides join keys on which join needs to be performed and merges the join results.
  *
- * @param <JOIN_KEY> type of the join key
+ * @param <JOIN_KEY>     type of the join key
  * @param <INPUT_RECORD> type of input records to be joined
- * @param <OUT> type of output object
+ * @param <OUT>          type of output object
  */
 @Beta
 public interface Joiner<JOIN_KEY, INPUT_RECORD, OUT> {
 
   /**
    * Return value for the join key on which join will be performed
    *
-   * @param stageName name of the stage to which records belongs to
+   * @param stageName   name of the stage to which records belongs to
    * @param inputRecord input record to be joined
    * @return returns join key
    * @throws Exception if there is some error getting the join key
+   * @deprecated getJoinKeys() should be used instead
```

Comment:
`use {@link #getJoinKeys()} instead`

that way the javadoc will link to the method.

Reply:
done